scanner/decisions/0009-nvd-cvss-scores.md

```diff
@@ -0,0 +1,121 @@
+      SEVERITY_IMPORTANT = 3;
+      SEVERITY_CRITICAL = 4;
+    }
+    message CVSS {
+      message V2 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      message V3 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      V2 v2 = 1;
+      V3 v3 = 2;
+    }
+    ...
+    string severity = 6;
+    ...
+    CVSS cvss = 12;
+  }
+  ...
+}
+```
+
+## Decision
+
+The VulnerabilityReport API will be extended to support more than one CVSS message. This will be done as follows:
+
+```
+message VulnerabilityReport {
+  message Vulnerability {
+    enum Severity {
+      SEVERITY_UNSPECIFIED = 0;
+      SEVERITY_LOW = 1;
+      SEVERITY_MODERATE = 2;
+      SEVERITY_IMPORTANT = 3;
+      SEVERITY_CRITICAL = 4;
+    }
+    message CVSS {
+      message V2 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      message V3 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      V2 v2 = 1;
+      V3 v3 = 2;
+    }
+    ...
+    string severity = 6;
+    ...
+    CVSS cvss = 12;
+    CVSS nvd_cvss = 13; <-- New field.
+  }
+  ...
+}
+```
+
+Currently, there is only a need to have CVSS scores from the vendor (if available) and NVD (if available), no more.
+Making this explicit, then, makes it easy and clear to users to find each scores and choose between the two.
+
+Another option is to provide a more generic solution. One way to do this would be to update `CVSS cvss = 12` to 
+`repeated CVSS cvss = 12` (note: the [proto3 spec](https://protobuf.dev/programming-guides/proto3/#updating) indicates this is compatible).
+Doing it this way gives us a more generic way of supporting multiple CVSS scores vs adding a new `nvd_cvss` field,
+and it may certainly be the way to go in the future should we want to support a variety of score sources; however,
+there is no need for a generic solution at this time, and adding the new field make it very easy to choose between the two options.
+
+### Handling RHSA/RHBA/RHEA
+
+Without loss of generality, RHSA/RHBA/RHEA will just be referred to as the more well-known RHSA variant of the three.
+
+Scanner V4 currently shows RHSA as the top-level entity, rather than the related CVE(s), when the CVE(s) is/are fixed.
+When this is done, Scanner V4 gives the RHSA the highest CVSS score from the associated CVE(s). We acknowledge this is not
+ideal, and there are plans to resolve this in the future. For now, we will need to support NVD scores in a compatible manner.
+
+The [`claircore.Vulnerability.Severity`](https://github.com/quay/claircore/blob/v1.5.25/vulnerability.go#L24) is currently set to the following:
+
+`severity=<severity>&cvss3_score=<score3>&cvss3_vector=<vector3>cvss2_score=<score2>&cvss2_vector=<vector2>`
+
+This URL encoding will be extended to include `cve=<CVE ID>`. 
+By doing this, Scanner V4 has the ability to relate the RHSA back to the CVE which has the highest score and search NVD for that CVE's score.
+
+## Consequences
```

Comment:
I am missing the discussion of the following consequences

1. Extending the severity field to encode the CVE related to the Advisory in the RHEL updater will impact our decision to remove the custom RHEL updater when the time comes to adopt the CSAF/VEX updater.
1. The impact on the other security sources that don't present a top-level CVE name and use advisories.
1. The impact on the existing CVSS field's behavior when the security source (e.g., OSV) does not offer them CVSS scores. Will we return empty and set NVD CVSS scores in the new field (or mapping if we change the decision), or keep the current behavior?

Reply:
Will update, thanks